fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

```diff
@@ -804,19 +868,31 @@ public void modifyTableColocate(Database db, OlapTable table, String colocateGro
                         }
                     }
                 }
+                // we also need to check the schema consistency with colocate group in other database
```

Comment:
It should be checked before the partition check.

Reply:
Why? I think if the table wants to join a new cg, and it doesn't belong to other cg before, the self check is necessary and should be put before checking colocation with cg in other db.